utils/aws/route53/__init__.py

```diff
@@ -0,0 +1,318 @@
+
+
+class Account(object):
+    """
+    Account represents an account and its DNS Zones
+
+    :param name: unique name that describes the account
+    :type name: str
+    """
+    def __init__(self, name):
+        self._name = name
+        self._zones = []
+
+    @property
+    def name(self):
+        """Returns the name of the account"""
+        return self._name
+
+    @property
+    def zones(self):
+        """
+        Returns the DNS zones managed under the account
+
+        :return: list of DNS zones
+        :rtype: list of Zone
+        """
+        return {a.name: a for a in self._zones}
+
+    def add_zone(self, zone):
+        """
+        Add a zone to the account
+
+        :param zone: DNS zone to add
+        :type zone: Zone
+        """
+        if self.zones.get(zone.name):
+            raise DuplicateException(zone, self)
+        self._zones.append(zone)
+
+    def get_zone(self, name):
+        """
+        Retrieve a zone by name
+
+        :param name: Zone name to retrieve
+        :type name: str
+        :return: DNS zone
+        :rtype: Zone
+        """
+        return self.zones.get(name)
+
+    def __repr__(self):
+        return f"Account<{self.name}>"
+
+
+class Zone(object):
+    """
+    Zone represents a DNS zone
+
+    :param name: DNS domain name
+    :param data: DNS zone data
+    :type name: str
+    :type data: dict
+    """
+    def __init__(self, name, data={}):
+        self._name = name.lower().rstrip('.')
+        self._records = []
+        self._data = data
+
+    @property
+    def data(self):
+        """Returns the zone data"""
+        return self._data
+
+    @property
+    def name(self):
+        """Returns the zone name"""
+        return self._name
+
+    @property
+    def records(self):
+        """Return DNS records"""
+        return {f"{r.name}": r for r in self._records}
+
+    def add_record(self, record):
+        """
+        Add a record to the zone
+
+        :param record: the record to add
+        :type record: Record
+        :raises DuplicateException: if a record by that name already exists
+        """
+        if self.records.get(record.name):
+            raise DuplicateException(record, self)
+        self._records.append(record)
+
+    def __eq__(self, other):
+        if not isinstance(other, Zone):
+            return False
+        return self.name == other.name
+
+    def __hash__(self):
+        return f"{self.name}".__hash__()
+
+    def __repr__(self):
+        return f"Zone<{self.name}>"
+
+
+class Record(object):
+    """
+    Record represents a DNS record
+
+    :param zone: the zone the record belongs to
+    :param name: the record name
+    :param data: the record data
+    :param awsdata: the original aws data for a dns record
+    :type zone: Zone
+    :type name: str
+    :type data: dict
+    :type awsdata: dict
+    """
+
+    SUPPORTED_TYPES = ['A', 'CNAME', 'NS', 'TXT']
+
+    def __init__(self, zone, name, data={}, awsdata={}):
```

Comment:
Same here... no mutable objects please.

Reply:
fixed in e4925c8d7c39c3a52523bc0482b9458b7f47c04b